bitcoin/src/bip32.rs

```diff
@@ -112,40 +112,76 @@ pub struct Xpub {
 #[cfg(feature = "serde")]
 crate::serde_utils::serde_string_impl!(Xpub, "a BIP-32 extended public key");
 
+/// A non-hardened child number.
+///
+/// # Note
+///
+/// Key index, within [0, 2^31 - 1].
+#[derive(Copy, Clone, PartialEq, Eq, Debug, PartialOrd, Ord, Hash)]
+pub struct NormalChildNumber(pub u32);
+
+/// A hardened child number
+///
+/// # Note
+///
+/// Key index, within [0, 2^31 - 1].
+#[derive(Copy, Clone, PartialEq, Eq, Debug, PartialOrd, Ord, Hash)]
+pub struct HardenedChildNumber(pub u32);
+
+impl From<NormalChildNumber> for ChildNumber {
+    fn from(child: NormalChildNumber) -> Self { ChildNumber::Normal(child) }
+}
+
+impl From<ChildNumber> for NormalChildNumber {
+    fn from(child: ChildNumber) -> Self {
+        match child {
+            ChildNumber::Normal(normal) => normal,
+            _ => panic!("ChildNumber is not normal"),
```

Comment:
I'm not sure about this.
Suggestions?

Reply:
`From` defiinitely should not panic. Since the conversion is fallible we should implement `TryFrom` instead if we want a conversion function.